demos/text_detection_demo/src/text_detection.cpp

```diff
@@ -98,7 +98,36 @@ std::vector<cv::RotatedRect> maskToBoxes(const cv::Mat &mask, float min_area, fl
     }
 
     return bboxes;
-  }
+}
+
+std::vector<cv::RotatedRect> coordToBoxes(const std::vector<float> &coords,
+                                          float min_area, float min_height,
+                                          const cv::Size &input_shape,
+                                          const cv::Size &image_size) {
+    std::vector<cv::RotatedRect> bboxes;
+    int max_bbox_idx = coords.size() / 5;
+    float x_scale = image_size.width / float(input_shape.width);
+    float y_scale = image_size.height / float(input_shape.height);
+
+    for (int i = 1; i <= max_bbox_idx; i++) {
+        std::vector<float> prediction(coords.begin() + (i - 1) * 5,
```

Comment:
Seems like `const float *prediction` would achieve the same effect with less code and without unnecessarily copying the data.

Reply:
Thank you for your detailed and very useful feedback and even additional explanation. But here I don't really understand how float pointer could replace the vector with 5  float values, which is created on 113 line. Did you meant 115 line maybe?
` float confidence = prediction[4];`

Reply:
All you do with `prediction` is index it, so as far as I can tell, you could replace it with `const float *prediction = &coords[(i - 1) * 5];` and the rest of the code would work the same way.